### PR TITLE
JNG-4220 add_add_method_for_builder

### PR DESCRIPTION
**Pull request by Madmat8**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4220" title="JNG-4220" target="_blank">Sub-taskJNG-4220</a>  Builder method for embedded collection add support of add and remove methods
  </td></table>
  <br />
 

JNG-4220_add_add_method_for_builder

### CODE REVIEW

**Review by gaborflorian — APPROVED** [review]

File: src/main/java/hu/blackbelt/structured/map/proxy/util/ReflectionUtil.java [hunk 1]
```diff
@@ -36,6 +36,7 @@ public final class ReflectionUtil {
 
     private static final String GET_PREFIX = "get";
     private static final String SET_PREFIX = "set";
+    private static final String ADD_PREFIX = "addTo";
```
Comment (noherczeg) [inline src/main/java/hu/blackbelt/structured/map/proxy/util/ReflectionUtil.java:39]: What about "remove"?

File: src/main/java/hu/blackbelt/structured/map/proxy/MapProxy.java [hunk 2]
```diff
@@ -706,6 +708,82 @@ private boolean invokeEquals(Object proxy, Object[] args) throws InvocationTarge
         return false;
     }
 
+    public void invokeAddOrRemove(Method m, Object[] args, boolean add) throws ExecutionException {
```
Comment (noherczeg) [inline src/main/java/hu/blackbelt/structured/map/proxy/MapProxy.java:711]: This method has to split into more parts. Method / blocks with names make it easier to understand the code. Also helps debugging.

**Review by noherczeg — CHANGES_REQUESTED** [review]

File: src/main/java/hu/blackbelt/structured/map/proxy/util/ReflectionUtil.java (hunk 1 above)
Comment (Madmat8) [reply src/main/java/hu/blackbelt/structured/map/proxy/util/ReflectionUtil.java:39]: We discussed with @robertcsakany that addTo... is enough for the builder.

File: src/main/java/hu/blackbelt/structured/map/proxy/MapProxy.java (hunk 2 above)
Comment (Madmat8) [reply src/main/java/hu/blackbelt/structured/map/proxy/MapProxy.java:711]: This is part of [my previous PR](https://github.com/BlackBeltTechnology/structured-map-proxy/pull/82). I originally created separate functions for addto and removefrom, but there are minimal differences between addto and removefrom, so @robertcsakany asked to merge them.

**Review by bencegelei — APPROVED** [review]

**Review by noherczeg — APPROVED** [review]

**Review by robertcsakany — APPROVED** [review]

**Pull request merged**